model_compression_toolkit/core/pytorch/pytorch_device_config.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import torch
+
+from model_compression_toolkit import logger
+from model_compression_toolkit.core.pytorch.constants import CUDA, CPU
+
+
+class DeviceManager:
+    """
+    A singleton class to manage the PyTorch device (CPU or CUDA) across different parts of the project.
+    Ensures that only one instance of this class is created and used throughout all files.
+    """
+
+    _instance = None
+
+    def __new__(cls):
+        """
+        Override the __new__ method to implement the singleton pattern.
+        Ensures that only one instance of DeviceManager is created.
+        """
+        if cls._instance is None:
+            cls._instance = super(DeviceManager, cls).__new__(cls)
+            # Initialize the default device
+            cls._instance.DEVICE = torch.device(CUDA if torch.cuda.is_available() else CPU)
+        return cls._instance
+
+    def set_device(self, device_name: str):
+        """
+        Set the device for PyTorch operations.
+
+        Args:
+            device_name (str): The name of the device, e.g., 'cuda:0' or 'cpu'.
+
+        If the specified device is not valid or available, it prints an error message without changing the current device.
```

Comment:
Do you think we should alert or raise an error? If just a warning - fix the comment.

Reply:
I think a warning. Changed the comment.